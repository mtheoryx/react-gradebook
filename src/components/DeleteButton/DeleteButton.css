.DeleteButton {
	height: 40px;
	width: 40px;
	border: 1px solid #ccc;
	background-color: #fafafa;
	position: relative;
	cursor: pointer;
}
.DeleteButton:hover {
	background-color: #ececec;
	border-color: #7b7b7b;

}
.DeleteButton:active {
	background-color: #dadada;
	border-color: #6a6a6a;
}
.DeleteButton:active,
.DeleteButton:focus {
	outline: none;
}
.DeleteButton:after {
	content: '\00d7';
	font-size: 24px;
	color: #505050;
}
