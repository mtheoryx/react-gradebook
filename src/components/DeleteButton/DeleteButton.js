import React from 'react';
import './DeleteButton.css';

const DeleteButton = props => {
    const { onDelete, itemId } = props;

    return (
        <button className="deleteButton" onClick={ () => onDelete(itemId) }>&times;</button>
    )
};

DeleteButton.propTypes = {
    onDelete: React.PropTypes.func.isRequired,
    itemId: React.PropTypes.string.isRequired
};

export default DeleteButton;


