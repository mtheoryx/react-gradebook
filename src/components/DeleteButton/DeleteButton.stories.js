import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { WithNotes } from '@kadira/storybook-addon-notes';

import DeleteButton from './DeleteButton';
import maxWidthCentered from '../../helpers/storybook-helpers/max-width-centered';

const deleteButtonNote = `
    This positioning of the X inside is pretty poor and should be fixed.
`;

storiesOf('DeleteButton', module)
    .addDecorator(maxWidthCentered)
    .add('Default', () => (
        <WithNotes notes={ deleteButtonNote } >
            <DeleteButton onDelete={ action('delete-button-click') } itemId={'student1'} />
        </WithNotes>
    ));
