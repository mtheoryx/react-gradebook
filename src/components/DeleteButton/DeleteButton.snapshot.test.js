import React from 'react';
import renderer from 'react-test-renderer';
import DeleteButton from './DeleteButton';

test('Delete button renders correctly', () => {
    const onDelete = () => {};
    const itemId = 'item1';
    const rendered = renderer.create(
        <DeleteButton onDelete={onDelete} itemId={itemId}/>
    );

    expect(rendered.toJSON()).toMatchSnapshot();
});
