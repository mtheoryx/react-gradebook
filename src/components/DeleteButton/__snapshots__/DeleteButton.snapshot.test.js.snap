exports[`test Delete button renders correctly 1`] = `
<button
  className="DeleteButton"
  onClick={[Function]}>
  Â 
</button>
`;
